src/shims/windows/handle.rs

```diff
@@ -0,0 +1,162 @@
+use rustc_target::abi::HasDataLayout;
+use std::mem::variant_count;
+
+use crate::*;
+
+/// A Windows `HANDLE` that represents a resource instead of being null or a pseudohandle.
```

Comment:
The windows foreign_items still return arbitrary integers for handles in GetStdHandle and GetProcessHeap. Is that a problem? I did it just to get the tests to pass.^^

Reply:
For `GetProcessHeap` it's not a problem because its std only, so it doesn't have to be generally correct. For `GetStdHandle`, it's a problem but I think it can be ignored until fs support is added for windows.

Reply:
Fair. Please add comments in those shims explaining your reasoning.